### PR TITLE
zephyr: espressif: modify SOC_FAMILY according to new HWMv2

### PR DESCRIPTION
**Pull request by sylvioalves**
Update FAMILY reference as needed.

### CODE REVIEW

**Comment by sylvioalves** [comment]
@almir-okato @dleach02 PTAL.

**Comment by sylvioalves** [comment]
@d3zd3z Is it possible I get permission to be reviewer in here as well?

File: boot/zephyr/Kconfig [hunk 1]
```diff
@@ -722,7 +722,7 @@ config MULTITHREADING
 	default y if BOOT_SERIAL_CDC_ACM #usb driver requires MULTITHREADING
 	default y if BOOT_USB_DFU_GPIO || BOOT_USB_DFU_WAIT
 	default n if SOC_FAMILY_NRF
-	default n if SOC_FAMILY_ESP32 && MCUBOOT
+	default n if SOC_FAMILY_ESPRESSIF_ESP32 && MCUBOOT
```
Comment (sylvioalves) [inline boot/zephyr/Kconfig:725]: @almir-okato, not related to this PR but, do we need `&& MCUBOOT` in there? This is mcuboot build, so `MCUBOOT` will always be set I think.

Comment (almir-okato) [reply boot/zephyr/Kconfig:725]: I think you can remove the `&& MCUBOOT` part

**Pull request merged**